src/Aspire.Dashboard/Components/Layout/MobileNavMenu.razor.cs

```diff
@@ -0,0 +1,99 @@
+    public required IStringLocalizer<Resources.Layout> Loc { get; init; }
+
+    [Inject]
+    public required IJSRuntime JS { get; init; }
+
+    private Task NavigateToAsync(string url)
+    {
+        NavigationManager.NavigateTo(url);
+        return Task.CompletedTask;
+    }
+
+    private IEnumerable<MobileNavMenuEntry> GetMobileNavMenuEntries()
+    {
+        if (DashboardClient.IsEnabled)
+        {
+            yield return new MobileNavMenuEntry(
+                Loc[nameof(Resources.Layout.NavMenuResourcesTab)],
+                () => NavigateToAsync(DashboardUrls.ResourcesUrl()),
+                DesktopNavMenu.ResourcesIcon(),
+                LinkMatchRegex: new Regex($"^{DashboardUrls.ResourcesUrl()}$")
+            );
+
+            yield return new MobileNavMenuEntry(
+                Loc[nameof(Resources.Layout.NavMenuConsoleLogsTab)],
+                () => NavigateToAsync(DashboardUrls.ConsoleLogsUrl()),
+                DesktopNavMenu.ConsoleLogsIcon(),
+                LinkMatchRegex: GetNonIndexPageRegex(DashboardUrls.ConsoleLogsUrl())
+            );
+        }
+
+        yield return new MobileNavMenuEntry(
+            Loc[nameof(Resources.Layout.NavMenuStructuredLogsTab)],
+            () => NavigateToAsync(DashboardUrls.StructuredLogsUrl()),
+            DesktopNavMenu.StructuredLogsIcon(),
+            LinkMatchRegex: GetNonIndexPageRegex(DashboardUrls.StructuredLogsUrl())
+        );
+
+        yield return new MobileNavMenuEntry(
+            Loc[nameof(Resources.Layout.NavMenuTracesTab)],
+            () => NavigateToAsync(DashboardUrls.TracesUrl()),
+            DesktopNavMenu.TracesIcon(),
+            LinkMatchRegex: GetNonIndexPageRegex(DashboardUrls.TracesUrl())
+        );
+
+        yield return new MobileNavMenuEntry(
+            Loc[nameof(Resources.Layout.NavMenuMetricsTab)],
+            () => NavigateToAsync(DashboardUrls.MetricsUrl()),
+            DesktopNavMenu.MetricsIcon(),
+            LinkMatchRegex: GetNonIndexPageRegex(DashboardUrls.MetricsUrl())
+        );
+
+        yield return new MobileNavMenuEntry(
+            Loc[nameof(Resources.Layout.MainLayoutAspireRepoLink)],
+            async () =>
+            {
+                await JS.InvokeVoidAsync("open", ["https://aka.ms/dotnet/aspire/repo", "_blank"]);
+            },
+            new AspireIcons.Size24.GitHub()
+        );
+
+        yield return new MobileNavMenuEntry(
+            Loc[nameof(Resources.Layout.MainLayoutAspireDashboardHelpLink)],
+            LaunchHelpAsync,
+            new Icons.Regular.Size24.QuestionCircle()
+        );
+
+        yield return new MobileNavMenuEntry(
+            Loc[nameof(Resources.Layout.MainLayoutLaunchSettings)],
+            LaunchSettingsAsync,
+            new Icons.Regular.Size24.Settings()
+        );
+    }
+
+    private static Regex GetNonIndexPageRegex(string pageRelativeBasePath)
+    {
+        return new Regex($"^({pageRelativeBasePath}|{pageRelativeBasePath}/.+)$");
```

Comment:
For safety, I think [Regex.Escape(String)](https://learn.microsoft.com/en-us/dotnet/api/system.text.regularexpressions.regex.escape?view=net-8.0) should be used on the path.
```suggestion
        pageRelativeBasePath = Regex.Escape(pageRelativeBasePath);
        return new Regex($"^({pageRelativeBasePath}|{pageRelativeBasePath}/.+)$");
```